Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2222,7 +2227,38 @@ ExceptionOr<Ref<ImageData>> CanvasRenderingContext2DBase::getImageData(int sx, i
 
     ASSERT(pixelBuffer->format().colorSpace == toDestinationColorSpace(computedColorSpace));
 
-    return { { ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull())) } };
+    auto imageData = ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull()));
+    
+#if ENABLE(NOISY_APIS)
+    if (DeprecatedGlobalSettings::noisyAPIsEnabled()) {
+        IdentifierProvider* identifierProvider = canvasBase().scriptExecutionContext()->identifierProvider();
+        // Using WeakRandom as I could explicitly set the seed.
+        // FIXME: set and use canvasIdentifier instead of domainIdentifier.
+        WeakRandom randomNumberGenerator = WeakRandom(identifierProvider->domainIdentifier());
+
+        WTFLogAlways("ID in getImageData() (seed for RNG): %u", randomNumberGenerator.seed());
+
+        auto& data = imageData->pixelBuffer()->data();
+        for (size_t i = 0; i < data.length(); i += 4) {
+            if (randomNumberGenerator.get() > 0.75) {
+                int fivePercentOffset = static_cast<int>(randomNumberGenerator.get() * 26 - 13);
+                size_t channel = i + static_cast<size_t>(randomNumberGenerator.get() * 3);
+                // WTFLogAlways("Random channel: %zu", channel);
```

Comment:
I know this is WIP, but please don't add any WTFLogAlways calls when the final PR is up.